homeassistant/components/sensor/growatt.py

```diff
@@ -0,0 +1,113 @@
+"""
+Support for Growatt Plant energy production sensors.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.growatt/
+"""
+import logging
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (CONF_USERNAME,
+                                 CONF_PASSWORD)
+from homeassistant.helpers.entity import Entity
+
+REQUIREMENTS = ['growatt==0.0.2']
+
+_LOGGER = logging.getLogger(__name__)
+
+UNIT = 'kWh'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+})
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the Growatt Plant sensor."""
+    username = config.get(CONF_USERNAME)
+    password = config.get(CONF_PASSWORD)
+
+    sensor_today = GrowattPlantToday(hass, username, password)
+    sensor_total = GrowattPlantTotal(hass, username, password)
+
+    add_entities([sensor_today, sensor_total])
+
+
+class GrowattPlant(Entity):
+    """Representation of a Growatt plant sensor."""
+
+    def __init__(self, hass, username, password):
+        """Initialize the sensor."""
+        self._hass = hass
+        self._unit_of_measurement = UNIT
+        self._state = None
+
+        import growatt
+        self._username = username
+        self._password = password
+        self.client = growatt.GrowattApi()
```

Comment:
This and `self.client.login(username, password)` should be moved to `setup_platform()`. If the credentials are wrong then the platform setup should fails as you can't recover from a login failure.

Reply:
Absolutely right.